symengine/parser.cpp

```diff
@@ -0,0 +1,184 @@
+#include <symengine/basic.h>
+#include <symengine/symbol.h>
+#include <symengine/add.h>
+#include <symengine/integer.h>
+#include <symengine/rational.h>
+#include <symengine/complex.h>
+#include <symengine/mul.h>
+#include <symengine/pow.h>
+#include <symengine/functions.h>
+#include <symengine/constants.h>
+#include <symengine/visitor.h>
+#include <vector>
+#include <stack>
+#include <map>
+#include <set>
+#include <string>
+
+namespace SymEngine {
+
+class expressionParser
+{
+    std::set<char> OPERATORS = {'-', '+', '*', '/', '^', '(', ')', ','};
+    std::map<char, int> op_precedence =   {
+                                    {')', 0}, {',', 0}, {'-', 1},  {'+', 1},
+                                    {'*', 3}, {'/', 4}, {'^', 5}
+                                    };
+    std::vector<int> operator_end;
+    std::string s;
+    unsigned int s_len;
+
+public:
+
+    RCP<const Basic> parse_string(unsigned int l, unsigned int h)
```

Comment:
Can you think of any case where it might be useful to expose this by making `parse_string` public? I didn't see this in the tests. If not, it would be better to keep it private.